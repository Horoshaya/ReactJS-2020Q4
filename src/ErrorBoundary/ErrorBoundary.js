import React from 'react';
import styles from './ErrorBoundary.css';

export default class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  componentDidCatch() {
    this.setState({ hasError: true });
  }

  render() {
    const { hasError } = this.state;
    const { children } = this.props;
    if (hasError) {
      return (
        <div className={styles.errorInfo}>
          Something went wrong. Please try again later.
        </div>
      );
    }
    return <>{children}</>;
  }
}
