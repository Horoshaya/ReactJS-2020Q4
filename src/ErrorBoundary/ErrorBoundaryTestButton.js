import React, { Component } from "react";
import styles from "./ErrorBoundaryTestButton.css";

export default class ErrorBoundaryTestButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: {
        name: "",
      },
    };

    this.isShowError = false;

    this.showError = this.showError.bind(this);
  }

  showError() {
    this.setState({ error: null });
    this.isShowError = true;
  }
  render() {
    return (
      <>
        <button
          className={styles.button}
          type="submit"
          onClick={this.showError}
        >
          Error Button
        </button>
        <div>{this.isShowError ? <p>{new Error("I crashed!")}</p> : true}</div>
      </>
    );
  }
}
