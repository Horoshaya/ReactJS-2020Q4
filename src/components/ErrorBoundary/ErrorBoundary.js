import React, { Component } from "react";
import styles from "./ErrorBoundary.css";

export default class ErrorBoundary extends Component {
  state = {
    hasError: false,
  };

  componentDidCatch(error, info) {
    console.log("error", error);
    console.log("info", info);
    this.setState({ hasError: true });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div className={styles.errorInfo}>
          <p>Извините, произошла ошибка!</p>
          <p>Мы пытаемся сделать все возможное, чтобы ее исправить</p>
        </div>
      );
    } else {
      return this.props.children;
    }
  }
}
