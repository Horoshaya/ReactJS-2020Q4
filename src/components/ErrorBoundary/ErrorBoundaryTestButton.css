.button {
  padding: 10px 15px;
  background: rgb(167, 47, 47);
  color: #ffffff;
  border: none;
  border-radius: 5px;
  cursor: pointer;
}
