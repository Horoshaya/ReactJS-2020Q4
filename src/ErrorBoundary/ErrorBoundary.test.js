import React from 'react';
import rerender from 'react-test-renderer';
import ErrorBoundary from './ErrorBoundary';

describe('when Wrapper rendered', () => {
  test('then child component should be rendered', () => {
    const TestChildComponent = (
      <div className="error">Something went wrong. Please try again later.</div>
    );
    const component = rerender.create(
      <ErrorBoundary state={true}>{TestChildComponent}</ErrorBoundary>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
