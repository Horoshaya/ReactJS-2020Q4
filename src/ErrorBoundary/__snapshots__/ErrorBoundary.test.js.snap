// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Wrapper rendered then child component should be rendered 1`] = `
<div
  className="error"
>
  Something went wrong. Please try again later.
</div>
`;
